ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/events/AlterTableHandler.java

```diff
@@ -92,21 +93,46 @@ private Scenario scenarioType(org.apache.hadoop.hive.metastore.api.Table before,
     }
   }
 
-  // return true, if event needs to be dumped, else return false.
-  private boolean handleForTableLevelReplication(Context withinContext) {
-    String oldName = before.getTableName();
-    String newName = after.getTableName();
+  private boolean isSetForBootstrapByReplaceHandler(Context withinContext, String tblName) {
+    return (withinContext.oldReplScope != null)
+            && !(ReplUtils.tableIncludedInReplScope(withinContext.oldReplScope, tblName))
+            && (ReplUtils.tableIncludedInReplScope(withinContext.replScope, tblName));
+  }
+
+  // Return true, if event needs to be dumped, else return false.
+  private boolean handleForTableLevelReplication(Context withinContext, String tblName) {
+    // For alter, if the table does not satisfy the new policy then ignore the event. In case of replace
+    // policy, if the table does not satisfy the old policy, then ignore the event. As, if the table satisfy the new
+    // policy, then the table will be bootstrapped by replace handler anb if the table does not satisfy the new policy,
+    // then anyways the table should be ignored. The event should be ignored if the table is already in the list of
+    // tables to be bootstrapped.
+    if (!ReplUtils.tableIncludedInReplScope(withinContext.replScope, tblName)) {
+      // In case of replace, it will be dropped during load. In normal case just ignore the alter event.
+      LOG.debug("Table " + tblName + " does not satisfy the policy");
+      return false;
+    } else if ((withinContext.getTablesForBootstrap().contains(tblName.toLowerCase()))
```

Comment:
I think, the else if flow is dead code as we skip this event from caller itself. Utils.shouldReplicate method returns false for this case. Pls check.

Reply:
done